janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/rest/RestElasticSearchClient.java

```diff
@@ -108,12 +108,17 @@
 
     private final boolean esVersion7;
 
+    private Integer retryOnConflict;
+
+    private final String retryOnConflictKey;
+
     public RestElasticSearchClient(RestClient delegate, int scrollKeepAlive, boolean useMappingTypesForES7) {
         this.delegate = delegate;
         majorVersion = getMajorVersion();
         this.scrollKeepAlive = scrollKeepAlive+"s";
         esVersion7 = ElasticMajorVersion.SEVEN.equals(majorVersion);
         useMappingTypes = majorVersion.getValue() < 7 || (useMappingTypesForES7 && esVersion7);
+        retryOnConflictKey = majorVersion.getValue() >= 7 ? "retry_on_conflict" : "_retry_on_conflict";
```

Comment:
I don't see that underscore is needed for elasticsearch 6. Could you point me to ES documentation which shows that? Maybe I am missing something.
Below both documentations shows that the correct parameter is `retry_on_conflict` and not `_retry_on_conflict`.
https://www.elastic.co/guide/en/elasticsearch/reference/6.2/docs-bulk.html
https://www.elastic.co/guide/en/elasticsearch/reference/6.4/docs-update.html

Reply:
https://www.elastic.co/guide/en/elasticsearch/reference/6.0/docs-bulk.html

Reply:
That is interesting. In version `6.0` the parameter is `_retry_on_conflict`, in version `6.1` and above the parameter is `retry_on_conflict`. As I understand it is breaking change (which is really strange). I will try `_retry_on_conflict` parameter with ES version 6.x later today. Maybe it works also. 
The thing is that we are checking only major version here. If we should use `_retry_on_conflict` for `6.0` and `retry_on_conflict` for `6.1` then we would need to get and compare `feature version` also.

Reply:
OK. I've checked this parameter. In the versions `6.x` bigger than 6.0 both `_retry_on_conflict` and `retry_on_conflict` works identically, so this isn't a breaking change.